### PR TITLE
Migrate to modern stack, including TS, new linting, updated APIs

### PR DESCRIPTION
**Pull request by janishutz**
# What's new
- Migrated to TS
- All methods feature full JSDoc documentation
- Add new APIs with Promise and other modern JS features
- Test have been updated to work with the new APIs too
- Updated ESLint config that also does code formatting

# Why
I have used this project before, but only browser based, where the current stack was good enough, but now that I am using this project for a web-app with a bundler, where I like to use NPM packages, it simply wasn't enough any more.

### CODE REVIEW

**Comment by simplePCBuilding** [comment]
Alright, I have finished this. I haven't bothered implementing tests for the new APIs, as they are the same as the old ones, but simply using modern browser APIs.

**Comment by simplePCBuilding** [comment]
@lokesh Sorry for the large PR. It's a lot of changes in the package-lock.json and the built files

**Comment by simplePCBuilding** [comment]
I would also like to suggest adding the wiki feature of GitHub. You can let me know when it is up, I can edit the Wiki for you and create docs. Alternatively, I could write some docs as a website

**Comment by simplePCBuilding** [comment]
I might just publish my own version of this to npm, so both versions exist

**Comment by janishutz** [comment]
I did publish my own package. `npm i @janishutz/color-thief`

**Comment by challet** [comment]
> I did publish my own package. `npm i @janishutz/color-thief`

Your package is actually `@janishutz/colorthief` ;)

**Comment by janishutz** [comment]
Ohh right... thanks for pointing this out!

File: README.md [hunk 1]
```diff
@@ -1,13 +1,44 @@
...
+    <img alt="GitHub Repo stars" src="https://img.shields.io/github/stars/janishutz/color-thief">
+    <img alt="GitHub watchers" src="https://img.shields.io/github/watchers/janishutz/color-thief">
+    <img alt="GitHub forks" src="https://img.shields.io/github/forks/janishutz/color-thief">
+    <img alt="GitHub commit activity" src="https://img.shields.io/github/commit-activity/m/janishutz/color-thief">
+    <br>
+    <img alt="GitHub all releases" src="https://img.shields.io/github/downloads/janishutz/color-thief/total?label=Downloads (total)">
+    <img alt="GitHub release (latest by date)" src="https://img.shields.io/github/downloads/janishutz/color-thief/latest/total?label=Downloads (latest)">
+    <img alt="Latest release" src="https://img.shields.io/github/release/janishutz/color-thief.svg">
+    <img alt="App Version" src="https://img.shields.io/github/package-json/v/janishutz/color-thief.svg?label=Development Version">
+</div>
```
Comment (sillyangel) [inline README.md:22]: all of this is pointing to the wrong project

File: README.md [hunk 2]
```diff
@@ -1,13 +1,44 @@
...
 
-### View the [demo page](https://lokeshdhakar.com/projects/color-thief/) for examples, API docs, and more.
+Full TypeScript implementation of the package originally created by [@lokesh.dhakar](https://github.com/lokesh/colorthief)
+
+**View the [demo page](https://lokeshdhakar.com/projects/color-thief/) for examples, API docs, and more.**
+
+# Installation
+Install this package from NPM or download specific files from the releases page!
+
+```
```
Comment (sillyangel) [inline README.md:33]: linking to the wrong npm package

File: package.json [hunk 3]
```diff
@@ -1,12 +1,12 @@
 {
-    "name": "colorthief",
-    "version": "2.4.0",
+    "name": "@janishutz/colorthief",
```
Comment (sillyangel) [inline package.json:2]: wrong package name, and version?

File: package.json [hunk 4]
```diff
@@ -1,12 +1,12 @@
...
+    "version": "3.0.1",
     "author": {
-        "name": "Lokesh Dhakar",
-        "email": "lokesh.dhakar@gmail.com",
-        "url": "http://lokeshdhakar.com/"
+        "name": "Janis Hutz",
```
Comment (sillyangel) [inline package.json:5]: wrong author data

File: package.json [hunk 5]
```diff
@@ -16,40 +16,50 @@
         "photo",
         "canvas"
     ],
-    "homepage": "http://lokeshdhakar.com/projects/color-thief/",
+    "type": "module",
+    "homepage": "https://lokeshdhakar.com/projects/color-thief/",
     "repository": {
         "type": "git",
-        "url": "https://github.com/lokesh/color-thief.git"
+        "url": "https://github.com/janishutz/color-thief.git"
```
Comment (sillyangel) [inline package.json:23]: pointing to the wrong repository url

**Review by sillyangel — CHANGES_REQUESTED** [review]
some of this is pointing to the wrong url's, and other data in package.json

**Comment by janishutz** [comment]
I should probably close this PR, running my own package with these features, as it seems that here the JS-only version will remain

**Comment by sillyangel** [comment]
> I should probably close this PR, running my own package with these features, as it seems that here the JS-only version will remain

I agree

**Pull request closed without merging**